src/R/Components/Impl/ConnectionManager/Commands/ReconnectCommand.cs

```diff
@@ -21,12 +21,12 @@ public class ReconnectCommand : IAsyncCommand {
             ? CommandStatus.Supported
             : CommandStatus.SupportedAndEnabled;
 
-        public Task<CommandResult> InvokeAsync() {
```

Comment:
I mean, if we want to have real async command execution, we could add `InvokeAsync` and `GetStatusAsync` or make  new `ICommandTargetAsync` and do `.Wait` only when sending down to OLE or base editor. However, OLE shim and base calls would have to transition to UI thread.  So up to you.